types/schemaoptions.d.ts

```diff
@@ -7,7 +7,10 @@ declare module 'mongoose' {
     currentTime?: () => (NativeDate | number);
   }
 
-  interface SchemaOptions {
+  type TypeKeyBaseType = string;
```

Comment:
I really like the custom type key support, this is great :+1:

Reply:
Thanks